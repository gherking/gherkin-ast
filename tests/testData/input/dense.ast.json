{
  "gherkinDocument": {
    "feature": {
      "tags": [
        {
          "location": {
            "line": 3,
            "column": 1
          },
          "name": "@tag1",
          "id": "c7ca7255-b99f-413b-b45a-28a8a6e91a22"
        },
        {
          "location": {
            "line": 3,
            "column": 7
          },
          "name": "@tag2",
          "id": "0eb408b6-5aa4-446c-ae5e-6748f11ffab0"
        },
        {
          "location": {
            "line": 5,
            "column": 1
          },
          "name": "@tag3",
          "id": "864bceda-bcf0-4819-a17e-d91eb8e8cc99"
        }
      ],
      "location": {
        "line": 7,
        "column": 1
      },
      "language": "en",
      "keyword": "Feature",
      "name": "Name",
      "description": "  Feature description\n  Other fekature description",
      "children": [
        {
          "rule": {
            "id": "b5e85f3f-9cb7-412f-a6ce-8a04ef490b0e",
            "location": {
              "line": 14,
              "column": 3
            },
            "keyword": "Rule",
            "name": "Name",
            "description": "    Rule description\n    Other rule description",
            "children": [
              {
                "background": {
                  "id": "a6a2de2d-1d82-428b-a9b6-c14d0587c71e",
                  "location": {
                    "line": 19,
                    "column": 5
                  },
                  "keyword": "Background",
                  "name": "Name",
                  "description": "      Background description\n      Other background description",
                  "steps": [
                    {
                      "id": "3d66c80e-020c-4500-981f-546e9c5dc577",
                      "location": {
                        "line": 24,
                        "column": 7
                      },
                      "keyword": "Given ",
                      "text": "step"
                    },
                    {
                      "id": "7a07649a-c17c-4e7a-aee1-adeda9017d18",
                      "location": {
                        "line": 26,
                        "column": 7
                      },
                      "keyword": "When ",
                      "text": "step"
                    },
                    {
                      "id": "01138581-c9f0-4cdf-982a-56d40b5b1ada",
                      "location": {
                        "line": 28,
                        "column": 7
                      },
                      "keyword": "Then ",
                      "text": "step"
                    }
                  ]
                }
              },
              {
                "scenario": {
                  "id": "d95a0940-a845-40c5-b09c-73ef95dc9cf7",
                  "tags": [
                    {
                      "location": {
                        "line": 30,
                        "column": 5
                      },
                      "name": "@tag1",
                      "id": "f363394a-920c-4972-85e9-8d25e3c8ba70"
                    },
                    {
                      "location": {
                        "line": 30,
                        "column": 11
                      },
                      "name": "@tag2",
                      "id": "8d3c9609-cc09-42e6-80f5-037f02192ac3"
                    }
                  ],
                  "location": {
                    "line": 32,
                    "column": 5
                  },
                  "keyword": "Scenario",
                  "name": "Name",
                  "description": "      Scenario description\n      Other scenario description",
                  "steps": [
                    {
                      "id": "d65152ca-14c5-4956-a25f-fb18257a8deb",
                      "location": {
                        "line": 37,
                        "column": 7
                      },
                      "keyword": "Given ",
                      "text": "step",
                      "docString": {
                        "location": {
                          "line": 39,
                          "column": 9
                        },
                        "content": "docstring\nother docstring",
                        "delimiter": "\"\"\""
                      }
                    },
                    {
                      "id": "14af7cd5-1159-47a2-87c0-0097d5a1ab9c",
                      "location": {
                        "line": 44,
                        "column": 7
                      },
                      "keyword": "When ",
                      "text": "step",
                      "dataTable": {
                        "location": {
                          "line": 46,
                          "column": 9
                        },
                        "rows": [
                          {
                            "id": "ce3354b8-dccd-4758-9f26-8bd921b366f3",
                            "location": {
                              "line": 46,
                              "column": 9
                            },
                            "cells": [
                              {
                                "location": {
                                  "line": 46,
                                  "column": 11
                                },
                                "value": "data"
                              },
                              {
                                "location": {
                                  "line": 46,
                                  "column": 19
                                },
                                "value": "table"
                              }
                            ]
                          },
                          {
                            "id": "b952c875-51bd-40a5-bd60-b5fb6ee17fab",
                            "location": {
                              "line": 48,
                              "column": 9
                            },
                            "cells": [
                              {
                                "location": {
                                  "line": 48,
                                  "column": 11
                                },
                                "value": "value"
                              },
                              {
                                "location": {
                                  "line": 48,
                                  "column": 19
                                },
                                "value": "value"
                              }
                            ]
                          },
                          {
                            "id": "55fa4134-8978-4281-a8ed-23de45e3673b",
                            "location": {
                              "line": 49,
                              "column": 9
                            },
                            "cells": [
                              {
                                "location": {
                                  "line": 49,
                                  "column": 11
                                },
                                "value": "value"
                              },
                              {
                                "location": {
                                  "line": 49,
                                  "column": 19
                                },
                                "value": "value"
                              }
                            ]
                          }
                        ]
                      }
                    },
                    {
                      "id": "d8d32cac-c1a4-4126-908d-b93c0395f8ce",
                      "location": {
                        "line": 51,
                        "column": 7
                      },
                      "keyword": "Then ",
                      "text": "step",
                      "docString": {
                        "location": {
                          "line": 52,
                          "column": 9
                        },
                        "content": "title\n=====\ndocstring with content type",
                        "delimiter": "\"\"\"",
                        "mediaType": "markdown"
                      }
                    },
                    {
                      "id": "9a7d8cc1-9f6e-46c0-8ced-2631e8364a27",
                      "location": {
                        "line": 57,
                        "column": 7
                      },
                      "keyword": "And ",
                      "text": "step",
                      "docString": {
                        "location": {
                          "line": 58,
                          "column": 5
                        },
                        "content": "docstring with backtick and content type",
                        "delimiter": "```",
                        "mediaType": "markdown"
                      }
                    }
                  ],
                  "examples": []
                }
              },
              {
                "scenario": {
                  "id": "50530d0a-b053-44a5-b177-f561922cd53e",
                  "tags": [
                    {
                      "location": {
                        "line": 62,
                        "column": 5
                      },
                      "name": "@tag1",
                      "id": "d5992186-137f-4ea9-a70d-48fd49447e11"
                    },
                    {
                      "location": {
                        "line": 62,
                        "column": 11
                      },
                      "name": "@tag2",
                      "id": "dc65b080-27d5-418c-af84-965975555f28"
                    }
                  ],
                  "location": {
                    "line": 64,
                    "column": 5
                  },
                  "keyword": "Scenario Outline",
                  "name": "Name",
                  "description": "      Scenario outline description\n      Other scenario outline description",
                  "steps": [
                    {
                      "id": "f1e2254b-1fb8-4ba5-9117-961a16af62bf",
                      "location": {
                        "line": 68,
                        "column": 7
                      },
                      "keyword": "Given ",
                      "text": "step <v>"
                    },
                    {
                      "id": "fa4d24b6-0674-4a81-88fa-2bb4c050a173",
                      "location": {
                        "line": 69,
                        "column": 7
                      },
                      "keyword": "When ",
                      "text": "step <w>"
                    }
                  ],
                  "examples": [
                    {
                      "id": "9753dec4-69c8-4fd9-8e21-d5e2268b88a7",
                      "tags": [
                        {
                          "location": {
                            "line": 71,
                            "column": 7
                          },
                          "name": "@tag1",
                          "id": "5d553e7e-dbb2-4a8e-92ab-9a7e8d7b6df7"
                        },
                        {
                          "location": {
                            "line": 71,
                            "column": 13
                          },
                          "name": "@tag2",
                          "id": "f3dfacff-3f1d-4618-b28c-b8cec0b39e03"
                        }
                      ],
                      "location": {
                        "line": 73,
                        "column": 7
                      },
                      "keyword": "Examples",
                      "name": "Name",
                      "description": "",
                      "tableHeader": {
                        "id": "c8abc7e0-0839-46cb-8cf0-d360e8de134f",
                        "location": {
                          "line": 75,
                          "column": 9
                        },
                        "cells": [
                          {
                            "location": {
                              "line": 75,
                              "column": 11
                            },
                            "value": "v"
                          },
                          {
                            "location": {
                              "line": 75,
                              "column": 15
                            },
                            "value": "w"
                          }
                        ]
                      },
                      "tableBody": [
                        {
                          "id": "9f376802-fcf7-486f-a0d6-05c5191c0a77",
                          "location": {
                            "line": 77,
                            "column": 9
                          },
                          "cells": [
                            {
                              "location": {
                                "line": 77,
                                "column": 11
                              },
                              "value": "1"
                            },
                            {
                              "location": {
                                "line": 77,
                                "column": 15
                              },
                              "value": "2"
                            }
                          ]
                        },
                        {
                          "id": "e8e84479-2b23-4c52-9894-56573e5dc542",
                          "location": {
                            "line": 78,
                            "column": 9
                          },
                          "cells": [
                            {
                              "location": {
                                "line": 78,
                                "column": 11
                              },
                              "value": "2"
                            },
                            {
                              "location": {
                                "line": 78,
                                "column": 15
                              },
                              "value": "3"
                            }
                          ]
                        }
                      ]
                    }
                  ]
                }
              }
            ],
            "tags": [
              {
                "location": {
                  "line": 12,
                  "column": 3
                },
                "name": "@tag1",
                "id": "73a435e3-5b9f-4355-abd0-b8c8875d2d36"
              },
              {
                "location": {
                  "line": 12,
                  "column": 9
                },
                "name": "@tag2",
                "id": "edf6f16b-317a-45c8-8371-d9f4138f39d7"
              }
            ]
          }
        }
      ]
    },
    "comments": [
      {
        "location": {
          "line": 1,
          "column": 1
        },
        "text": "# Comment(s) at the beginning of the file"
      },
      {
        "location": {
          "line": 2,
          "column": 1
        },
        "text": "# Comment(s) of all tags"
      },
      {
        "location": {
          "line": 4,
          "column": 1
        },
        "text": "# Comment(s) of @tag3"
      },
      {
        "location": {
          "line": 6,
          "column": 1
        },
        "text": "# Comment(s) of the feature"
      },
      {
        "location": {
          "line": 10,
          "column": 1
        },
        "text": "  # Additional comment(s) of the feature"
      },
      {
        "location": {
          "line": 11,
          "column": 1
        },
        "text": "  # Comment(s) of all tags"
      },
      {
        "location": {
          "line": 13,
          "column": 1
        },
        "text": "  # Comment(s) of the rule"
      },
      {
        "location": {
          "line": 17,
          "column": 1
        },
        "text": "    # Additional comment(s) of the rule"
      },
      {
        "location": {
          "line": 18,
          "column": 1
        },
        "text": "    # Comment(s) of the background"
      },
      {
        "location": {
          "line": 22,
          "column": 1
        },
        "text": "      # Additional comment(s) of the background"
      },
      {
        "location": {
          "line": 23,
          "column": 1
        },
        "text": "      # Comment(s) of the given step"
      },
      {
        "location": {
          "line": 25,
          "column": 1
        },
        "text": "      # Comment(s) of the when step"
      },
      {
        "location": {
          "line": 27,
          "column": 1
        },
        "text": "      # Comment(s) of the then step"
      },
      {
        "location": {
          "line": 29,
          "column": 1
        },
        "text": "    # Comment(s) of all tags"
      },
      {
        "location": {
          "line": 31,
          "column": 1
        },
        "text": "    # Comment(s) of the scenario"
      },
      {
        "location": {
          "line": 35,
          "column": 1
        },
        "text": "      # Additional comment(s) of the scenario"
      },
      {
        "location": {
          "line": 36,
          "column": 1
        },
        "text": "      # Comment(s) of the given step"
      },
      {
        "location": {
          "line": 38,
          "column": 1
        },
        "text": "        # Comment(s) of the docstirng"
      },
      {
        "location": {
          "line": 43,
          "column": 1
        },
        "text": "      # Comment(s) of the when step"
      },
      {
        "location": {
          "line": 45,
          "column": 1
        },
        "text": "        # Comment(s) of the data table header (row)"
      },
      {
        "location": {
          "line": 47,
          "column": 1
        },
        "text": "        # Comment(s) of a data table row"
      },
      {
        "location": {
          "line": 50,
          "column": 1
        },
        "text": "      # Comment(s) of the then step"
      },
      {
        "location": {
          "line": 61,
          "column": 1
        },
        "text": "    # Comment(s) of all tags"
      },
      {
        "location": {
          "line": 63,
          "column": 1
        },
        "text": "    # Comment(s) of the scenario outline"
      },
      {
        "location": {
          "line": 67,
          "column": 1
        },
        "text": "      # Another comment(s) of the scenario outline"
      },
      {
        "location": {
          "line": 70,
          "column": 1
        },
        "text": "      # Comment(s) of all tags"
      },
      {
        "location": {
          "line": 72,
          "column": 1
        },
        "text": "      # Comment(s) of the examples"
      },
      {
        "location": {
          "line": 74,
          "column": 1
        },
        "text": "        # Comment(s) of the examples table"
      },
      {
        "location": {
          "line": 76,
          "column": 1
        },
        "text": "        # Comment(s) of an examples table row"
      },
      {
        "location": {
          "line": 79,
          "column": 1
        },
        "text": "  # General comment(s)"
      }
    ],
    "uri": "tests/data/comment.feature"
  }
}