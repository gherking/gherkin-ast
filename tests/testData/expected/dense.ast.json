{
  "_id": "a270db0d-cd33-4028-9dd5-78fe399dd26d",
  "uri": "tests/data/comment.feature",
  "sourceFile": "comment.feature",
  "sourceFolder": "tests/data",
  "targetFile": "comment.feature",
  "targetFolder": "tests/data",
  "feature": {
    "_id": "d4a97d0c-3f7b-4595-be8d-7513dda4cbe9",
    "language": "en",
    "keyword": "Feature",
    "name": "Name",
    "description": "Feature description\nOther feature description",
    "elements": [
      {
        "_id": "33fc43bd-e448-43a0-a60b-a7506111d8ad",
        "keyword": "Rule",
        "name": "Name",
        "description": "Rule description\nOther rule description",
        "elements": [
          {
            "_id": "c65e258c-4035-4d70-8637-883019d0ce6f",
            "keyword": "Background",
            "name": "Name",
            "description": "Background description\nOther background description",
            "steps": [
              {
                "_id": "f824e76a-645a-4de5-9a74-a8b269d0907b",
                "keyword": "Given",
                "text": "step",
                "dataTable": null,
                "docString": null,
                "comment": {
                  "_id": "030da24d-0459-485b-a46b-3e361c244696",
                  "text": "# Additional comment(s) of the background\n# Comment(s) of the given step"
                }
              },
              {
                "_id": "40a2413d-9555-4800-a740-0bbeaa7050d7",
                "keyword": "When",
                "text": "step",
                "dataTable": null,
                "docString": null,
                "comment": {
                  "_id": "ed127194-1017-42fd-bdd7-75223edd8d7f",
                  "text": "# Comment(s) of the when step"
                }
              },
              {
                "_id": "1ab8c4e8-3f3e-42cb-b274-627ad667b961",
                "keyword": "Then",
                "text": "step",
                "dataTable": null,
                "docString": null,
                "comment": {
                  "_id": "273ac038-e39a-4305-be0f-c129a6ed0a43",
                  "text": "# Comment(s) of the then step"
                }
              }
            ],
            "precedingComment": {
              "_id": "067f927b-9bea-4e50-afd6-5896ef8f3cdb",
              "text": "# Additional comment(s) of the rule\n# Comment(s) of the background"
            },
            "descriptionComment": null
          },
          {
            "_id": "08c13de2-3a61-4457-99a4-d3d9bd44d388",
            "keyword": "Scenario",
            "name": "Name",
            "description": "Scenario description\nOther scenario description",
            "steps": [
              {
                "_id": "8d68ccb5-13c6-4800-9b15-f12ef754e43d",
                "keyword": "Given",
                "text": "step",
                "dataTable": null,
                "docString": {
                  "_id": "eb7d4cb9-bb57-46a1-bb16-3259a22821f3",
                  "content": "docstring\nother docstring",
                  "delimiter": "\"\"\"",
                  "mediaType": null,
                  "comment": {
                    "_id": "1343768f-0612-4984-a2d4-9037d980bce6",
                    "text": "# Comment(s) of the docstirng"
                  }
                },
                "comment": {
                  "_id": "b570d390-3dc3-4b66-b17e-b17588aaded3",
                  "text": "# Additional comment(s) of the scenario\n# Comment(s) of the given step"
                }
              },
              {
                "_id": "703f6d1e-8f11-451c-a8b1-6d31e439849b",
                "keyword": "When",
                "text": "step",
                "dataTable": {
                  "_id": "d052706c-f96d-4330-805d-d1881808af44",
                  "rows": [
                    {
                      "_id": "08a3770f-0848-4d89-9fc6-b9f152996f02",
                      "cells": [
                        {
                          "_id": "f071d630-88aa-44c7-acbf-c3190234f30b",
                          "value": "data"
                        },
                        {
                          "_id": "e016bbf8-3bc3-4215-a1e6-bef72717a7e2",
                          "value": "table"
                        }
                      ],
                      "comment": {
                        "_id": "413f6d8d-fee2-459f-a952-c3e721084144",
                        "text": "# Comment(s) of the data table header (row)"
                      }
                    },
                    {
                      "_id": "cc56bc78-d2e4-445f-a37a-7c3d0c735074",
                      "cells": [
                        {
                          "_id": "1a10e0f8-cee1-4cdf-9a21-d48567d8254c",
                          "value": "value"
                        },
                        {
                          "_id": "fc8e99e3-b3ec-4ebe-bcc4-e077f3d289b1",
                          "value": "value"
                        }
                      ],
                      "comment": {
                        "_id": "a836d28a-6cfe-40a6-b9c0-033d02cc2c5d",
                        "text": "# Comment(s) of a data table row"
                      }
                    },
                    {
                      "_id": "34dcd995-81e7-4298-9295-35a4f47127d8",
                      "cells": [
                        {
                          "_id": "a4203c35-40a5-431f-8554-ca8ffc7fdade",
                          "value": "value"
                        },
                        {
                          "_id": "a874e8e3-8fcc-433b-8476-b91a6e527f8e",
                          "value": "value"
                        }
                      ],
                      "comment": null
                    }
                  ]
                },
                "docString": null,
                "comment": {
                  "_id": "638e694c-03fc-498d-a84a-09222da58542",
                  "text": "# Comment(s) of the when step"
                }
              },
              {
                "_id": "671108b8-c5d3-4189-9406-2a5324705cdc",
                "keyword": "Then",
                "text": "step",
                "dataTable": null,
                "docString": {
                  "_id": "3b8375b5-3f6f-47d0-a93c-2f2ddab0238a",
                  "content": "title\n=====\ndocstring with content type",
                  "delimiter": "\"\"\"",
                  "mediaType": "markdown",
                  "comment": null
                },
                "comment": {
                  "_id": "d53869b4-21e7-4e85-a8dc-9c603eaaffe7",
                  "text": "# Comment(s) of the then step"
                }
              },
              {
                "_id": "35060afd-dda3-4923-85b6-9cde274b68b4",
                "keyword": "And",
                "text": "step",
                "dataTable": null,
                "docString": {
                  "_id": "5327436a-25f7-4c44-9ca0-6d9e9a1cd3cd",
                  "content": "docstring with backtick and content type",
                  "delimiter": "```",
                  "mediaType": "markdown",
                  "comment": null
                },
                "comment": null
              }
            ],
            "precedingComment": {
              "_id": "50cd6040-8b5e-44a5-898e-8cf7a5d09ef7",
              "text": "# Comment(s) of the scenario"
            },
            "descriptionComment": null,
            "tags": [
              {
                "_id": "ef73117d-0f4d-45cc-a169-0889ba02a995",
                "name": "tag1",
                "comment": null
              },
              {
                "_id": "7672cede-32e9-4f1e-8080-8dc34822393a",
                "name": "tag2",
                "comment": null
              }
            ],
            "tagComment": {
              "_id": "c5ec6904-a93f-4d89-af44-695d3d57704e",
              "text": "# Comment(s) of all tags"
            }
          },
          {
            "_id": "a378067e-cebf-493e-bf19-1d2763ebfa01",
            "keyword": "Scenario Outline",
            "name": "Name",
            "description": "Scenario outline description\nOther scenario outline description",
            "steps": [
              {
                "_id": "88d4bcde-8cd6-41d9-b5ae-67ce3fc15e3b",
                "keyword": "Given",
                "text": "step <v>",
                "dataTable": null,
                "docString": null,
                "comment": {
                  "_id": "12876f52-3e48-4917-9eb4-4970cb571bb7",
                  "text": "# Another comment(s) of the scenario outline"
                }
              },
              {
                "_id": "3a83d737-6042-42fc-82eb-c4ea6dd6924e",
                "keyword": "When",
                "text": "step <w>",
                "dataTable": null,
                "docString": null,
                "comment": null
              }
            ],
            "precedingComment": {
              "_id": "441ddcd9-e9b1-451d-b469-227b0fa5751b",
              "text": "# Comment(s) of the scenario outline"
            },
            "descriptionComment": null,
            "tags": [
              {
                "_id": "3321b300-c735-4a13-9601-83f533d9e2cc",
                "name": "tag1",
                "comment": null
              },
              {
                "_id": "ca1d1840-5163-4225-8d63-7c1fad81ba1d",
                "name": "tag2",
                "comment": null
              }
            ],
            "examples": [
              {
                "_id": "3f6d8528-2930-4cec-8b4b-126b8d7c49e0",
                "keyword": "Examples",
                "name": "Name",
                "header": {
                  "_id": "fda786ab-905c-4164-a9b7-52d093cfe8f6",
                  "cells": [
                    {
                      "_id": "d7c1609a-f4e1-403a-a7ee-bbbcddbed41b",
                      "value": "v"
                    },
                    {
                      "_id": "83ac3796-2768-4686-90b3-7b76cdda93fc",
                      "value": "w"
                    }
                  ],
                  "comment": {
                    "_id": "94bc32f7-2bda-431f-899e-d55c7702f226",
                    "text": "# Comment(s) of the examples table"
                  }
                },
                "precedingComment": {
                  "_id": "5617b7d9-a667-4116-ac4e-74a9d25ece37",
                  "text": "# Comment(s) of the examples"
                },
                "tags": [
                  {
                    "_id": "76695400-c573-4051-98c1-0bca1793e054",
                    "name": "tag1",
                    "comment": null
                  },
                  {
                    "_id": "fb910eb3-9c8b-41f8-881a-d13e8efdfb20",
                    "name": "tag2",
                    "comment": null
                  }
                ],
                "body": [
                  {
                    "_id": "03deadb9-b09b-4fa5-ab3e-0a9e324210d1",
                    "cells": [
                      {
                        "_id": "56144c57-3839-4154-a496-08fa86a51493",
                        "value": "1"
                      },
                      {
                        "_id": "f3872e77-8785-4049-8133-e445273c3be0",
                        "value": "2"
                      }
                    ],
                    "comment": {
                      "_id": "ca5de6cf-4900-40f2-b5b4-370bca6e3c42",
                      "text": "# Comment(s) of an examples table row"
                    }
                  },
                  {
                    "_id": "402458ac-b81c-44c0-8028-53efeac75abc",
                    "cells": [
                      {
                        "_id": "31272e28-d06d-4f05-aee1-f1de60c9d92b",
                        "value": "2"
                      },
                      {
                        "_id": "5d61e946-d442-4c07-a84d-43e15e9c31ee",
                        "value": "3"
                      }
                    ],
                    "comment": null
                  }
                ],
                "tagComment": {
                  "_id": "cf6f6a45-4a37-4606-9795-1c753ff2f421",
                  "text": "# Comment(s) of all tags"
                }
              }
            ],
            "tagComment": {
              "_id": "0c4112c3-7151-4186-852a-d245f2a40577",
              "text": "# Comment(s) of all tags"
            }
          }
        ],
        "tags": [
          {
            "_id": "17e587b9-5ee3-4d6e-b14e-97ae9dda9dc4",
            "name": "tag1",
            "comment": null
          },
          {
            "_id": "c2aacdd8-0c75-4d44-871c-531758624ace",
            "name": "tag2",
            "comment": null
          }
        ],
        "precedingComment": {
          "_id": "531afffa-75bb-4a9d-9df0-1be726ee7088",
          "text": "# Comment(s) of the rule"
        },
        "tagComment": {
          "_id": "417cdcf0-ce49-4fc1-aadf-5d4120cacd22",
          "text": "# Additional comment(s) of the feature\n# Comment(s) of all tags"
        },
        "descriptionComment": null
      }
    ],
    "tags": [
      {
        "_id": "baf3ec6b-766e-4716-8684-5b65b47ba3d0",
        "name": "tag1",
        "comment": null
      },
      {
        "_id": "20a9e481-5e15-4abf-87c2-537cd7d94e8e",
        "name": "tag2",
        "comment": null
      },
      {
        "_id": "b8dbef8e-2e17-4f9c-bc04-dc64617df58e",
        "name": "tag3",
        "comment": {
          "_id": "84bb2e6e-06f2-4609-95c8-0fa1f4b1c472",
          "text": "# Comment(s) of @tag3"
        }
      }
    ],
    "precedingComment": {
      "_id": "87b976b8-00cf-4b86-a0cf-86c6886afd01",
      "text": "# Comment(s) of the feature"
    },
    "tagComment": {
      "_id": "39fc6108-a875-4d75-a1e2-189846814880",
      "text": "# Comment(s) at the beginning of the file\n# Comment(s) of all tags"
    },
    "descriptionComment": null
  },
  "startComment": null,
  "endComment": {
    "_id": "f10e1acc-a197-466c-9844-01aba2a76d7f",
    "text": "# General comment(s)"
  }
}